linode/terraform-provider-linode
# new: Add `user_type` to user-related resources and data sources

## 📝 Description

This change adds a new `user_type` field to the following resources:

* `linode_user`
* `data.linode_user`
* `data.linode_users`

## ✔️ How to Test

The following tests assume you have pulled down this PR locally and are using a `LINODE_TOKEN` and `LINODE_URL` an account with children.

### Integration Tests

```
make int-test PKG_NAME=linode/user
make int-test PKG_NAME=linode/users
```

### Manual Testing

1. In a Terraform provider sandbox environment, apply the following configuration:

```terraform

resource "random_string" "name" {
  length = 12
  special = false
}

resource "linode_user" "test" {
  username = random_string.name.result
  email = "${random_string.name.result}@linode.com"
}

data "linode_user" "test" {
  depends_on = [linode_user.test]

  username = linode_user.test.username
}

data "linode_users" "test" {
  depends_on = [linode_user.test]

  filter {
    name = "username"
    values = [linode_user.test.username]
  }
}
``` 

2. Ensure the `user_type` field is populated for all three resource instances in `terraform.tfstate`
